envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/groups/AccessLogFilterFactory.kt

```diff
@@ -0,0 +1,51 @@
+package pl.allegro.tech.servicemesh.envoycontrol.groups
+
+import com.google.re2j.Pattern
+import io.envoyproxy.envoy.api.v2.core.RuntimeUInt32
+import io.envoyproxy.envoy.config.filter.accesslog.v2.AccessLogFilter
+import io.envoyproxy.envoy.config.filter.accesslog.v2.ComparisonFilter
+import io.envoyproxy.envoy.config.filter.accesslog.v2.AccessLog.Builder
+import io.envoyproxy.envoy.config.filter.accesslog.v2.StatusCodeFilter.newBuilder
+
+class AccessLogFilterFactory {
+    private val operators: Array<ComparisonFilter.Op> = arrayOf(
+        ComparisonFilter.Op.LE, ComparisonFilter.Op.GE, ComparisonFilter.Op.EQ
+    )
+    private val delimiter: Char = ':'
+    private val statusCodeFilterPattern: Pattern = Pattern.compile(
+        """(${operators.joinToString("|")})$delimiter(\d{3})"""
```

Comment:
one more this. Pattern should looks like `^(${operators.joinToString("|")})$delimiter(\d{3})$` because now `testeq:400test` works

Reply:
Please change this.

Reply:
Changed, but can not agree that `testeq:400test`was matching. Pattern is validated for whole input https://github.com/google/re2j/blob/re2j-1.3/java/com/google/re2j/Pattern.java#L144